### PR TITLE
`r\application_insights` Add validation for `workspace_id`

### LINKED ISSUE #15576: azurerm_application_insight : optional workspace_id set to null when no value set as input.
Hello,

using following terraform version and providers : 

`tf_1.1 C:\...\Infras\IsolatedWeb - 5.1>terraform -v
Terraform v1.1.6
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.97.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0`

having the following resource : 

`resource "azurerm_application_insights" "my_application_insights" {
  name                = module.name.full_name
  location            = var.location
  tags                = var.tags
  resource_group_name = var.resource_group_name
  application_type    = "web"
}
`

The **workspace_id as been set outside of terraform,** so when terraform plan we have the following (but workspaci_id is not set as modified) : 

`Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the
last "terraform apply":

  # module.my_app_insight["westeurope_00"].azurerm_application_insights.my_application_insights has changed
  ~ resource "azurerm_application_insights" "my_application_insights" {
      + force_customer_storage_for_profiler   = false
        id                                    = "/subscriptions/bc046308-e9d7-4ad4-a8e6-de54fdb5d58f/resourceGroups/z-adf-csa-web-npd-ew1-rgp01/providers/Microsoft.Insights/components/z-adf-csa-web-npd-ew1-ain01"
        name                                  = "z-adf-csa-web-npd-ew1-ain01"
        tags                                  = {
            "global-app"       = "CSA"
            "global-dataclass" = "Confidential"
            "global-dcs"       = "DIO"
            "global-env"       = "NPD"
            "global-opco"      = "ADF"
            "global-project"   = "DAFR"
            "local-owner"      = "ads.application.support@axa-direct.com"
        }
        # (15 unchanged attributes hidden)
    }
`
a terraform plan will figure this output : 

`Terraform will perform the following actions:

  # module.my_app_insight["westeurope_00"].azurerm_application_insights.my_application_insights will be updated in-place
  ~ resource "azurerm_application_insights" "my_application_insights" {
        id                                    = "/subscriptions/bc046308-e9d7-4ad4-a8e6-de54fdb5d58f/resourceGroups/z-adf-csa-web-npd-ew1-rgp01/providers/Microsoft.Insights/components/z-adf-csa-web-npd-ew1-ain01"
        name                                  = "z-adf-csa-web-npd-ew1-ain01"
        tags                                  = {
            "global-app"       = "CSA"
            "global-dataclass" = "Confidential"
            "global-dcs"       = "DIO"
            "global-env"       = "NPD"
            "global-opco"      = "ADF"
            "global-project"   = "DAFR"
            "local-owner"      = "ads.application.support@axa-direct.com"
        }
      **_- workspace_id_**                          = "(_workspace_id_)" **-> null**
        # (15 unchanged attributes hidden)
    }
`
The workspace_id of the application insights want to be nullify, but no input for this workspace_id as been set.

looking at the azurerm_application_insight component at https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_insights
it appears that the workspace_id is optionnal, and not set by default to null.

could be fine that this value is not nullify if not set as input.

thank-you,

best regards,

Philippe

### PR DESCRIPTION
**Pull request by myc2h6o**
Fix #15576

### CODE REVIEW

File: website/docs/r/application_insights.html.markdown [hunk 1]
```diff
@@ -94,7 +94,9 @@ The following arguments are supported:
 
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 
-* `workspace_id` - (Optional) Specifies the id of a log analytics workspace resource
+* `workspace_id` - (Optional) Specifies the id of a log analytics workspace resource.
+
+~> **NOTE:** This can not be removed after set. More details can be found at [Migrate to workspace-based Application Insights resources](https://docs.microsoft.com/azure/azure-monitor/app/convert-classic-resource#migration-process)
```
Comment (tombuildsstuff) [inline website/docs/r/application_insights.html.markdown:99]: we should also add a CustomizeDiff to conditionally trigger this for a Workspace Application Insights instance

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @myc2h6o 

Thanks for this PR - taking a look through whilst this documents this limitation we should also add a CustomizeDiff to conditionally recreate this if this field is changed when using the Workspace model.

Thanks!

**Comment by myc2h6o** [comment]
Update comment to remove waiting-response tag

**Comment by myc2h6o** [comment]
Hi @tombuildsstuff `workspace_id` is likely to be interpolated so instead of adding a CustomizeDiff, I'm putting the validation inside the CreateUpdate instead, could you take a look?

**Comment by myc2h6o** [comment]
@katbyte probably it's better to just make it ForceNew, I've updated the change, could you take a look?

**Review by katbyte — APPROVED** [review]
Thanks @myc2h6o - LGTM ⛵

**Pull request merged**